src/rtos/rtos_freertos_stackings.c

```diff
@@ -44,637 +44,643 @@ Important Note: If you modify one of the stack_register_offset array, also check
-	{ XT_REG_IDX_AR36, -1, 32 },		/* A36 */
-	{ XT_REG_IDX_AR37, -1, 32 },		/* A37 */
-	{ XT_REG_IDX_AR38, -1, 32 },		/* A38 */
-	{ XT_REG_IDX_AR39, -1, 32 },		/* A39 */
-	{ XT_REG_IDX_AR40, -1, 32 },		/* A40 */
-	{ XT_REG_IDX_AR41, -1, 32 },		/* A41 */
-	{ XT_REG_IDX_AR42, -1, 32 },		/* A42 */
-	{ XT_REG_IDX_AR43, -1, 32 },		/* A43 */
-	{ XT_REG_IDX_AR44, -1, 32 },		/* A44 */
-	{ XT_REG_IDX_AR45, -1, 32 },		/* A45 */
-	{ XT_REG_IDX_AR46, -1, 32 },		/* A46 */
-	{ XT_REG_IDX_AR47, -1, 32 },		/* A47 */
-	{ XT_REG_IDX_AR48, -1, 32 },		/* A48 */
-	{ XT_REG_IDX_AR49, -1, 32 },		/* A49 */
-	{ XT_REG_IDX_AR50, -1, 32 },		/* A50 */
-	{ XT_REG_IDX_AR51, -1, 32 },		/* A51 */
-	{ XT_REG_IDX_AR52, -1, 32 },		/* A52 */
-	{ XT_REG_IDX_AR53, -1, 32 },		/* A53 */
-	{ XT_REG_IDX_AR54, -1, 32 },		/* A54 */
-	{ XT_REG_IDX_AR55, -1, 32 },		/* A55 */
-	{ XT_REG_IDX_AR56, -1, 32 },		/* A56 */
-	{ XT_REG_IDX_AR57, -1, 32 },		/* A57 */
-	{ XT_REG_IDX_AR58, -1, 32 },		/* A58 */
-	{ XT_REG_IDX_AR59, -1, 32 },		/* A59 */
-	{ XT_REG_IDX_AR60, -1, 32 },		/* A60 */
-	{ XT_REG_IDX_AR61, -1, 32 },		/* A61 */
-	{ XT_REG_IDX_AR62, -1, 32 },		/* A62 */
-	{ XT_REG_IDX_AR63, -1, 32 },		/* A63 */
-	{ XT_REG_IDX_LBEG, 0x58, 32 },		/* lbeg */
-	{ XT_REG_IDX_LEND, 0x5c, 32 },		/* lend */
-	{ XT_REG_IDX_LCOUNT, 0x60, 32 },		/* lcount */
-	{ XT_REG_IDX_SAR, 0x4c, 32 },		/* sar */
-	{ XT_REG_IDX_WINDOWBASE, -1, 32 },		/* windowbase */
-	{ XT_REG_IDX_WINDOWSTART, -1, 32 },		/* windowstart */
-	{ XT_REG_IDX_CONFIGID0, -1, 32 },		/* configid0 */
-	{ XT_REG_IDX_CONFIGID1, -1, 32 },		/* configid1 */
-	{ XT_REG_IDX_PS, 0x08, 32 },		/* ps */
-	{ XT_REG_IDX_THREADPTR, 0x70, 32 },		/* threadptr */
-	{ XT_REG_IDX_BR, 0x7c, 32 },		/* br */
-	{ XT_REG_IDX_SCOMPARE1, 0x80, 32 },		/* scompare1 */
-	{ XT_REG_IDX_ACCLO, 0x74, 32 },		/* acclo */
-	{ XT_REG_IDX_ACCHI, 0x78, 32 },		/* acchi */
-	{ XT_REG_IDX_M0, 0x84, 32 },		/* m0 */
-	{ XT_REG_IDX_M1, 0x88, 32 },		/* m1 */
-	{ XT_REG_IDX_M2, 0x8c, 32 },		/* m2 */
-	{ XT_REG_IDX_M3, 0x90, 32 },		/* m3 */
-	{ ESP32_REG_IDX_EXPSTATE, -1, 32 },		/* expstate */
-	{ ESP32_REG_IDX_F64R_LO, 0x94, 32 },		/* f64r_lo */
-	{ ESP32_REG_IDX_F64R_HI, 0x98, 32 },		/* f64r_hi */
-	{ ESP32_REG_IDX_F64S, 0x9c, 32 },		/* f64s */
-	{ XT_REG_IDX_F0, -1, 32 },		/* f0 */
-	{ XT_REG_IDX_F1, -1, 32 },		/* f1 */
-	{ XT_REG_IDX_F2, -1, 32 },		/* f2 */
-	{ XT_REG_IDX_F3, -1, 32 },		/* f3 */
-	{ XT_REG_IDX_F4, -1, 32 },		/* f4 */
-	{ XT_REG_IDX_F5, -1, 32 },		/* f5 */
-	{ XT_REG_IDX_F6, -1, 32 },		/* f6 */
-	{ XT_REG_IDX_F7, -1, 32 },		/* f7 */
-	{ XT_REG_IDX_F8, -1, 32 },		/* f8 */
-	{ XT_REG_IDX_F9, -1, 32 },		/* f9 */
-	{ XT_REG_IDX_F10, -1, 32 },		/* f10 */
-	{ XT_REG_IDX_F11, -1, 32 },		/* f11 */
-	{ XT_REG_IDX_F12, -1, 32 },		/* f12 */
-	{ XT_REG_IDX_F13, -1, 32 },		/* f13 */
-	{ XT_REG_IDX_F14, -1, 32 },		/* f14 */
-	{ XT_REG_IDX_F15, -1, 32 },		/* f15 */
-	{ XT_REG_IDX_FCR, -1, 32 },		/* fcr */
-	{ XT_REG_IDX_FSR, -1, 32 },		/* fsr */
+	{ 0,   0x04, 32 },		/* PC */
```

Comment:
@ianstcdns sequential reg number works for the stackings. Considering this list will not change, I don't think getting the reg_numbers during runtime will benefit us. Please comment.

Reply:
It's fine to do something like that on your branch, but it shouldn't be merged upstream as it would break sparse mappings.  Keep in mind that populating registers at runtime also has the benefit that you only have a single register list (in the core config file), so would be less error-prone.

Reply:
@ianstcdns I implemented draft dynamic stacking (meaning populating register numbers during runtime) It works ok but still need some work. You know when we create an XL patch they generally asked to split it up. So I would like to create several patches for nuttx OS first. One of the patch will include static register numbers same as in this  PR.

You have commented above
_It's fine to do something like that on your branch, but it shouldn't be merged upstream as it would break sparse mappings._
Can I ask why would our target_type would break the other configs? We don't provide the stacking arrays for generic xtensa. We are providing them for each chip separately. So we will not touch e.g. your `xtensa_chip.c` target.